### PR TITLE
[SPARK-49495][DOCS] Document and Feature Preview on the master branch via Live GitHub Pages Updates

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This pull request introduces functionalities that enable 'Document and Feature Preview on the master branch via Live GitHub Pages Updates'.

### Why are the changes needed?

retore
8861f0f9af3f397921ba1204cf4f76f4e20680bb
376382711e200aa978008b25630cc54271fd419b
58d73fe8e7cbff9878539d31430f819eff9fc7a1 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

https://github.com/yaooqinn/spark/actions/runs/10952355999

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+jobs:
+  docs:
+    name: Build and deploy documentation
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      pages: write
+    environment:
+      name: github-pages # https://github.com/actions/deploy-pages/issues/271
```
Comment (yaooqinn) [inline .github/workflows/pages.yml:39]: https://issues.apache.org/jira/browse/INFRA-26133?focusedCommentId=17883033&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17883033

**Review by HyukjinKwon — APPROVED** [review]
I am fine with this but wonder if we can make sure this is working before merging it in.

**Comment by yaooqinn** [comment]
Merged to master.

The website is now live. https://apache.github.io/spark/

BTW, This patch also brings the unexpected [action](https://github.com/apache/spark/actions/workflows/pages/pages-build-deployment) alive again. I will ask INFRA to disable it one more time

**Comment by dongjoon-hyun** [comment]
Ack. That's too bad. Ya, it would be great if we can recover it soon. 
Thank you, @yaooqinn .

**Comment by HyukjinKwon** [comment]
Thanks for doing this!

**Comment by yaooqinn** [comment]
The INFRA team has terminated these unexpected actions.

FYI,https://issues.apache.org/jira/browse/INFRA-26133

**Pull request closed without merging**